base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -261,17 +277,27 @@ pub struct LMDBDatabase {
     validator_nodes_mapping: DatabaseRef,
     /// Maps CodeTemplateRegistration <block_height, hash> -> TemplateRegistration
     template_registrations: DatabaseRef,
+    /// Stores a cache of the sparse merkle tree on the latest mod 1000 height
+    utxo_smt: DatabaseRef,
+
     _file_lock: Arc<File>,
     consensus_manager: ConsensusManager,
+    smt_cache_period: u64,
 }
 
 impl LMDBDatabase {
     pub fn new(
         store: &LMDBStore,
         file_lock: File,
         consensus_manager: ConsensusManager,
+        prune_interval: u64,
     ) -> Result<Self, ChainStorageError> {
         let env = store.env();
+        let smt_cache_period = if prune_interval == 0 {
+            SMT_CACHE_PERIOD
+        } else {
+            prune_interval / 2
+        };
```

Comment:
```suggestion
        let smt_cache_period = prune_interval.checked_div(2).unwrap_or(SMT_CACHE_PERIOD);
```

Reply:
if prune_interval == 0, it use `SMT_CACHE_PERIOD` 
0/2 = 0, not an invalid div